sdks/python/apache_beam/runners/portability/fn_api_runner/worker_handlers.py

```diff
@@ -737,11 +738,15 @@ def __init__(self,
 
   def host_from_worker(self):
     # type: () -> str
-    if sys.platform == "darwin":
+    if sys.platform == 'darwin':
       # See https://docs.docker.com/docker-for-mac/networking/
       return 'host.docker.internal'
-    else:
-      return super(DockerSdkWorkerHandler, self).host_from_worker()
+    if sys.platform != 'win32' and is_in_notebook():
```

Comment:
I'm not sure this is always necessary when running inside a notebook. I was able to run SqlTransform in a notebook for our demo last fall: https://gist.github.com/TheNeuralBit/9c79d71cbc90a962e795b80ca54fa3c8#file-simpler-python-pipelines-demo-ipynb

It seems like this is necessary in your case because the IPython kernel is running inside a docker container, but that's not always true. In my case I was running the kernel directly on my desktop. Is there something else we can test here?

Reply:
You're right, it's not necessary if the code is not running inside a container. But it still works even if the code is not inside a container.

The other thing we can check whether to enable this is through checking if the code is being executed in a docker container.
But it feels so clunky and fragile to check that: https://stackoverflow.com/questions/43878953/how-does-one-detect-if-one-is-running-within-a-docker-container-within-python/43879407

Reply:
If it works whether we're in a container or not, should we just do this all the time?

Reply:
There is some difference between accessing localhost and ip address: https://stackoverflow.com/questions/6938039/localhost-vs-real-ip-address.

For scenarios where localhost still works, we probably want to keep using localhost for better performance, depending how the user sets up their environment and runs pipelines. But for notebook environment, performance dip and Internet access are expected anyway, so we can safely enable this.